qemu/tests/ksm_ksmtuned.py

```diff
@@ -31,41 +32,76 @@ def run(test, params, env):
     :param params: Dictionary with test parameters.
     :param env: Dictionary with the test environment.
     """
-    def check_ksm(mem, stress=False):
+    def check_ksm(mem, threshold_reached=False):
         """
         :param mem: Boot guest with given memory, in KB
-        :param stress: Load stress or not
+        :ksmtuned_enabled: ksmtuned threshold is reached or not
         """
+        def heavyload_install():
+            if session.cmd_status(test_install_cmd) != 0:
+                logging.warning("Could not find installed heavyload in guest, "
+                                "will install it via winutils.iso ")
+                winutil_drive = utils_misc.get_winutils_vol(session)
+                if not winutil_drive:
+                    test.cancel("WIN_UTILS CDROM not found.")
+                install_cmd = params["install_cmd"] % winutil_drive
+                session.cmd(install_cmd)
+
+        def check_qemu_used_mem(qemu_pid, mem):
+            qemu_used_page = process.getoutput(get_qemu_used_mem % qemu_pid,
+                                               shell=True)
+            qemu_used_mem = float(qemu_used_page) * pagesize
+            if qemu_used_mem < mem * 0.95:
+                return False
+            return True
+
         params['mem'] = mem // 1024
         params['start_vm'] = 'yes'
         vm_name = params['main_vm']
         env_process.preprocess_vm(test, params, env, vm_name)
         vm = env.get_vm(vm_name)
-        vm.wait_for_login()
-        if stress and params["os_type"] == "linux":
+        session = vm.wait_for_login()
+        qemu_pid = vm.get_pid()
+        if params["os_type"] == "linux":
             params['stress_args'] = ('--cpu 4 --io 4 --vm 2 --vm-bytes %sM' %
                                      (int(params['mem']) // 2))
             stress_test = VMStress(vm, "stress", params)
             stress_test.load_stress_tool()
+        else:
+            install_path = params["install_path"]
+            test_install_cmd = 'dir "%s" | findstr /I heavyload' % install_path
+            heavyload_install()
+            heavyload_bin = r'"%s\heavyload.exe" ' % install_path
+            heavyload_options = ["/MEMORY 100", "/START"]
+            start_cmd = heavyload_bin + " ".join(heavyload_options)
+            stress_tool = BackgroundTest(session.cmd, (start_cmd,
+                                                       stress_timeout,
+                                                       stress_timeout))
+            stress_tool.start()
+            if not utils_misc.wait_for(stress_tool.is_alive, stress_timeout):
+                test.error("Failed to start heavyload process")
+        if not utils_misc.wait_for(lambda: check_qemu_used_mem(qemu_pid, mem),
+                                   stress_timeout, 10, 10):
+            test.error("QEMU used memory doesn't reach %sM in %ss"
+                       %(mem // 1024, stress_timeout))
```

Comment:
This is not match with L54: `mem*0.95`, suggest define a threshold

Reply:
Updated, thanks.